src/main/java/com/google/neighborgood/servlets/EditTaskServlet.java

```diff
@@ -66,12 +47,66 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
 
-    // Set the details and rewards to the newly input value
-    task.setProperty("detail", taskDetail);
-    task.setProperty("reward", rewardPts);
-    datastore.put(task);
+    // Edits tasks that have been claimed by setting the "helper" property to the userId
+    // of the helper and changing the task's status to "IN PROGRESS"
+    if (request.getParameter("action").equals("helpout")) {
+
+      UserService userService = UserServiceFactory.getUserService();
+      if (!userService.isUserLoggedIn()) {
+        System.err.println("User must be logged in to help out with a task");
+        response.sendError(
+            HttpServletResponse.SC_UNAUTHORIZED, "You must be logged in to help out with a task");
+      }
+
+      if (!task.getProperty("status").equals("OPEN")) {
+        System.err.println("Task must be open to be claimed by a helper");
+        response.sendError(
+            HttpServletResponse.SC_CONFLICT, "Task has already been claimed by another helper");
+      }
+
+      String userId = userService.getCurrentUser().getUserId();
+      task.setProperty("Helper", userId);
+      task.setProperty("status", "IN PROGRESS");
+      datastore.put(task);
+
+      response.setContentType("text/html;");
+      response.getWriter().println();
```

Comment:
If there is no input in the `println()` function, I think you can just remove the two lines of code.

Reply:
You're right, I'll just use a return statement in here instead, thanks!